_rules/auto-update-text-efbfc7.md

```diff
@@ -0,0 +1,419 @@
+
+```html
+<body onload="start()">
+	<p>Random number: <span id="target">1</span></p>
+	<div onclick="stop()" aria-label="">
+		<span style="background-color:#333; color:white; padding:5px">Stop updates</span>
+	</div>
+
+	<script type="text/javascript">
+		function change() {
+			var target = document.getElementById('target')
+			var number = Math.floor(Math.random() * 1000)
+			target.innerText = number
+		}
+
+		var updates
+		function start() {
+			updates = setInterval(change, 3000)
+		}
+
+		function stop() {
+			clearInterval(updates)
+		}
+	</script>
+</body>
+```
+
+#### Failed Example 5
+
+The text node automatically updates every 3 seconds after the page completes loading. The mechanism to stop the automatic updates has an accessible name that is only whitespace.
+
+```html
+<body onload="start()">
+	<p>Random number: <span id="target">1</span></p>
+	<div onclick="stop()" aria-label="   ">
+		<span style="background-color:#333; color:white; padding:5px">Stop updates</span>
+	</div>
+
+	<script type="text/javascript">
+		function change() {
+			var target = document.getElementById('target')
+			var number = Math.floor(Math.random() * 1000)
+			target.innerText = number
+		}
+
+		var updates
+		function start() {
+			updates = setInterval(change, 3000)
+		}
+
+		function stop() {
+			clearInterval(updates)
+		}
+	</script>
+</body>
+```
+
+#### Failed Example 6
+
+The text node (part of a progress bar) automatically updates every second after the page completes loading. There is no mechanism for the user to pause the auto-updates, other than waiting for them to end.
+
+```html
+<html>
+	<head>
+		<title>Example page for rule efbfc7</title>
+		<style>
+			.progress-wrap,
+			.progress-bar {
+				height: 20px;
+				max-width: 300px;
+				position: relative;
+			}
+
+			.progress-wrap {
+				background: #eee;
+			}
+
+			.progress-bar {
+				background: #cce0ff;
+				width: 0;
+				transition: width 1s;
+			}
+
+			.progress-text {
+				position: absolute;
+				top: 0;
+				width: 100%;
+				text-align: center;
+			}
+
+			.choices {
+				margin: 3em;
+			}
+		</style>
+	</head>
+
+	<body onload="start()">
+		<p>
+			You can pick a price you like, or wait for the final.
+		</p>
+		<div class="progress-wrap">
+			<div class="progress-bar" id="pb-demo"></div>
+			<div class="progress-text" id="pb-text">0%</div>
+		</div>
+		<div class="choices">
+			<button type="button" id="packa" onclick="select()">Pack A:</button>
+		</div>
+		<div>
+			<p id="outcome"></p>
+		</div>
+
+		<script type="text/javascript">
+			var n, updates, price
+
+			function updatePB() {
+				var bar = document.getElementById('pb-demo')
+				var text = document.getElementById('pb-text')
+				bar.style.width = n + '%'
+				text.innerHTML = n + '%'
+			}
+
+			function updateNum() {
+				var button = document.getElementById('packa')
+				if (n < 100) {
+					n = n + 10
+					updatePB(n)
```

Comment:
`updatePB` is using the global variable `n`. I'm in favour of adding a parameter to it (to reduce use of global variable), and thus keep this line as is, but currently it is not correct.